c_common/models/live_packet_gather/live_packet_gather.c

```diff
@@ -22,6 +22,12 @@ static circular_buffer without_payload_buffer;
 static circular_buffer with_payload_buffer;
 static bool processing_events = false;
 
+//! Provanence data store
+static uint32_t number_of_over_flows_none_payload = 0;
+static uint32_t number_of_over_flows_payload = 0;
```

Comment:
It might make sense to put the provenance data fields either:

a) In a struct 
b) In an array of uint32_ts indexed with the enum

A struct would have the benefit of removing the need for an enum and both options would simplify the code to copy it back into SDRAM.